providers/firetext/src/lib/firetext.provider.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+
+if (!globalThis.fetch) {
+  // eslint-disable-next-line global-require
+  globalThis.fetch = require('node-fetch');
+}
+
+export class FiretextSmsProvider implements ISmsProvider {
+  id = 'firetext';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private BASE_URL = 'https://www.firetext.co.uk/api/sendsms';
+
+  constructor(
+    private config: {
+      apiKey: string;
+      from: string;
+    }
+  ) {}
+
+  private parseResponse(body: string) {
+    const re = /^(\d+):(\d+)\s(.*)$/i;
+    const found = body.match(re);
+    const code = found?.[1] ?? 'Unknown status code';
+    const message = found?.[3] ?? 'Unknown status message';
+
+    return [code, message];
+  }
+
+  private parseHeaderId(headers: Headers) {
+    return headers.get('X-Message');
+  }
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const baseMessage = {
+      apiKey: this.config.apiKey,
+      to: options.to,
+      from: this.config.from,
+      message: options.content,
+    };
+
+    const urlSearchParams = new URLSearchParams(baseMessage);
+    const url = new URL(this.BASE_URL);
+    url.search = urlSearchParams.toString();
+
+    const response = await fetch(url.toString());
+    const body = await response.text();
+    const [code, message] = this.parseResponse(body);
+
+    if (code !== '0') {
+      throw new Error(`${code}: ${message}`);
+    }
+
+    const messageId = this.parseHeaderId(response.headers);
+
+    return {
+      id: messageId,
+      date: new Date().toISOString(),
```

Comment:
The date returned should be the date that the provider gives as return if it provides any. 
Setting the date to now at this moment after the response has been returned I think it is a bit misleading, as for cases where the request and the response have been slow, there could be a discrepancy between the time stored in the Provider database and the one we would show to Novu users. That could lead to more difficult debuggings for our users.
Does not Firetext returns a datetime in the response?

Reply:
@scopsy @davidsoderberg how bothering for our users this potential situation could be?

Reply:
Like I did this in the Sendgrid provider and do not see the minute/few minutes difference that could be if the provider always sends the webhook directly after the event have happened 😄  This is maybe something we need to discuss more with @scopsy and @oba2311...

Reply:
Just to mention I've added a fallback to `new Date()` - the `Date` header is set when I've tested but isn't documented and I'm not sure how much you can rely on it. Probably not a problem, but I'd rather return something rather than `undefined`.